flake on TestImageStreamImportRedHatRegistry integration test

I saw this earlier in my AWS devenv as well as in https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin/11001/consoleFull (testing https://github.com/openshift/origin/pull/6901)

```
Running TestImageStreamImportRedHatRegistry...
failed  TestImageStreamImportRedHatRegistry
I0216 13:18:03.688652   22266 plugins.go:62] Registered admission plugin "SecurityContextDeny"
I0216 13:18:03.688758   22266 plugins.go:62] Registered admission plugin "ResourceQuota"
I0216 13:18:03.688766   22266 plugins.go:62] Registered admission plugin "PersistentVolumeLabel"
I0216 13:18:03.688773   22266 plugins.go:62] Registered admission plugin "NamespaceLifecycle"
I0216 13:18:03.688778   22266 plugins.go:62] Registered admission plugin "NamespaceExists"
I0216 13:18:03.688784   22266 plugins.go:62] Registered admission plugin "NamespaceAutoProvision"
I0216 13:18:03.688790   22266 plugins.go:62] Registered admission plugin "LimitRanger"
I0216 13:18:03.688817   22266 plugins.go:62] Registered admission plugin "InitialResources"
I0216 13:18:03.688824   22266 plugins.go:62] Registered admission plugin "DenyEscalatingExec"
I0216 13:18:03.688828   22266 plugins.go:62] Registered admission plugin "DenyExecOnPrivileged"
I0216 13:18:03.688832   22266 plugins.go:62] Registered admission plugin "AlwaysDeny"
I0216 13:18:03.688838   22266 plugins.go:62] Registered admission plugin "AlwaysPullImages"
I0216 13:18:03.688842   22266 plugins.go:62] Registered admission plugin "AlwaysAdmit"
I0216 13:18:03.688854   22266 plugins.go:62] Registered admission plugin "SecurityContextConstraint"
I0216 13:18:03.688859   22266 plugins.go:62] Registered admission plugin "SCCExecRestrictions"
I0216 13:18:03.688866   22266 plugins.go:62] Registered admission plugin "RunOnceDuration"
I0216 13:18:03.688874   22266 plugins.go:62] Registered admission plugin "ClusterResourceOverride"
I0216 13:18:03.688880   22266 plugins.go:62] Registered admission plugin "ProjectRequestLimit"
I0216 13:18:03.688885   22266 plugins.go:62] Registered admission plugin "OriginPodNodeEnvironment"
I0216 13:18:03.688892   22266 plugins.go:62] Registered admission plugin "OriginNamespaceLifecycle"
I0216 13:18:03.688898   22266 plugins.go:62] Registered admission plugin "BuildByStrategy"
I0216 13:18:03.688904   22266 plugins.go:62] Registered admission plugin "BuildOverrides"
I0216 13:18:03.688909   22266 plugins.go:62] Registered admission plugin "BuildDefaults"
I0216 13:18:03.824448   22266 importer.go:308] importing remote Docker repository registry=https://registry.access.redhat.com repository=rhel7 insecure=false
I0216 13:18:04.157298   22266 importer.go:709] Registry responded to v2 Docker endpoint, but has no header for Docker Distribution https://registry.access.redhat.com/v2/: 404, http.Header{"Content-Type":[]string{"text/html; charset=UTF-8"}, "Date":[]string{"Tue, 16 Feb 2016 18:18:04 GMT"}, "Server":[]string{"Apache"}, "Content-Length":[]string{"238"}}
I0216 13:18:04.157374   22266 importer.go:312] unable to access repository &importer.importRepository{Ref:api.DockerImageReference{Registry:"registry.access.redhat.com", Namespace:"", Name:"rhel7", Tag:"", ID:""}, Registry:(*url.URL)(0xc2080c4480), Name:"rhel7", Insecure:false, Tags:[]importer.importTag{importer.importTag{Name:"latest", Image:(*api.Image)(nil), Err:error(nil)}}, Digests:[]importer.importDigest(nil), MaximumTags:0, AdditionalTags:[]string(nil), Err:error(nil)}: &importer.ErrNotV2Registry{Registry:"https://registry.access.redhat.com"}
I0216 13:18:04.157576   22266 importer.go:308] importing remote Docker repository registry=https://registry.access.redhat.com repository=rhel7 insecure=false
I0216 13:18:04.392215   22266 importer.go:709] Registry responded to v2 Docker endpoint, but has no header for Docker Distribution https://registry.access.redhat.com/v2/: 404, http.Header{"Date":[]string{"Tue, 16 Feb 2016 18:18:04 GMT"}, "Server":[]string{"Apache"}, "Content-Length":[]string{"238"}, "Content-Type":[]string{"text/html; charset=UTF-8"}}
I0216 13:18:04.392280   22266 importer.go:312] unable to access repository &importer.importRepository{Ref:api.DockerImageReference{Registry:"registry.access.redhat.com", Namespace:"", Name:"rhel7", Tag:"", ID:""}, Registry:(*url.URL)(0xc20807f500), Name:"rhel7", Insecure:false, Tags:[]importer.importTag{importer.importTag{Name:"latest", Image:(*api.Image)(nil), Err:error(nil)}}, Digests:[]importer.importDigest(nil), MaximumTags:0, AdditionalTags:[]string(nil), Err:error(nil)}: &importer.ErrNotV2Registry{Registry:"https://registry.access.redhat.com"}
I0216 13:18:04.392392   22266 client.go:406] Getting repository /rhel7 from {https  <nil> registry.access.redhat.com   }
--- FAIL: TestImageStreamImportRedHatRegistry-2 (1.53s)
	imageimporter_test.go:640: unexpected object: unversioned.Status{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ListMeta:unversioned.ListMeta{SelfLink:"", ResourceVersion:""}, Status:"Failure", Message:"Internal error occurred: the image \"07e361a1cd7cae70df4c585c2dbeceb25e069a01911c15a5d70d499eacd053ce\" in repository \"/rhel7\" was not found and may have been deleted", Reason:"InternalError", Details:(*unversioned.StatusDetails)(0xc20882d7c0), Code:500}
	imageimporter_test.go:641: imports: (*api.Image)(nil)
FAIL
```

May have been a blip in registry.access.redhat.com because it seems to work at this moment.

--- a/test/integration/imageimporter_test.go
+++ b/test/integration/imageimporter_test.go
@@ -19,6 +19,7 @@ import (
 	"k8s.io/kubernetes/pkg/api/errors"
 	"k8s.io/kubernetes/pkg/api/unversioned"
 	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	kerrors "k8s.io/kubernetes/pkg/util/errors"
 	"k8s.io/kubernetes/pkg/watch"
 
 	"github.com/openshift/origin/pkg/dockerregistry"
@@ -514,8 +515,22 @@ func TestImageStreamImportDockerHub(t *testing.T) {
 		},
 	}
 
-	i := importer.NewImageStreamImporter(importCtx, 3, nil)
-	if err := i.Import(gocontext.Background(), imports); err != nil {
+	err := retryWhenUnreachable(t, func() error {
+		i := importer.NewImageStreamImporter(importCtx, 3, nil)
+		if err := i.Import(gocontext.Background(), imports); err != nil {
+			return err
+		}
+
+		errs := []error{}
+		for i, d := range imports.Status.Images {
+			fromName := imports.Spec.Images[i].From.Name
+			if d.Status.Status != unversioned.StatusSuccess && fromName != "mysql/doesnotexistinanyform" {
+				errs = append(errs, fmt.Errorf("failed to import an image %s: %v", fromName, d.Status.Message))
+			}
+		}
+		return kerrors.NewAggregate(errs)
+	})
+	if err != nil {
 		t.Fatal(err)
 	}
 
@@ -547,16 +562,34 @@ func TestImageStreamImportQuayIO(t *testing.T) {
 	rt, _ := kclient.TransportFor(&kclient.Config{})
 	importCtx := importer.NewContext(rt, nil).WithCredentials(importer.NoCredentials)
 
+	repositoryName := quayRegistryName + "/coreos/etcd"
 	imports := &api.ImageStreamImport{
 		Spec: api.ImageStreamImportSpec{
 			Images: []api.ImageImportSpec{
-				{From: kapi.ObjectReference{Kind: "DockerImage", Name: "quay.io/coreos/etcd"}},
+				{From: kapi.ObjectReference{Kind: "DockerImage", Name: repositoryName}},
 			},
 		},
 	}
 
-	i := importer.NewImageStreamImporter(importCtx, 3, nil)
-	if err := i.Import(gocontext.Background(), imports); err != nil {
+	err := retryWhenUnreachable(t, func() error {
+		i := importer.NewImageStreamImporter(importCtx, 3, nil)
+		if err := i.Import(gocontext.Background(), imports); err != nil {
+			return err
+		}
+
+		errs := []error{}
+		for i, d := range imports.Status.Images {
+			fromName := imports.Spec.Images[i].From.Name
+			if d.Status.Status != unversioned.StatusSuccess {
+				if d.Status.Reason == "NotV2Registry" {
+					t.Skipf("the server did not report as a v2 registry: %#v", d.Status)
+				}
+				errs = append(errs, fmt.Errorf("failed to import an image %s: %v", fromName, d.Status.Message))
+			}
+		}
+		return kerrors.NewAggregate(errs)
+	}, imageNotFoundErrorPatterns...)
+	if err != nil {
 		t.Fatal(err)
 	}
 
@@ -567,20 +600,14 @@ func TestImageStreamImportQuayIO(t *testing.T) {
 		t.Fatalf("unexpected response: %#v", imports.Status.Images)
 	}
 	d := imports.Status.Images[0]
-	if d.Status.Status != unversioned.StatusSuccess {
-		if d.Status.Reason == "NotV2Registry" {
-			t.Skipf("the server did not report as a v2 registry: %#v", d.Status)
-		}
-		t.Fatalf("unexpected error: %#v", d.Status)
-	}
-	if d.Image == nil || len(d.Image.DockerImageManifest) == 0 || !strings.HasPrefix(d.Image.DockerImageReference, "quay.io/coreos/etcd@") || len(d.Image.DockerImageMetadata.ID) == 0 || len(d.Image.DockerImageLayers) == 0 {
-		t.Errorf("unexpected object: %#v", d.Image)
+	if d.Image == nil || len(d.Image.DockerImageManifest) == 0 || !strings.HasPrefix(d.Image.DockerImageReference, repositoryName+"@") || len(d.Image.DockerImageMetadata.ID) == 0 || len(d.Image.DockerImageLayers) == 0 {
 		s := spew.ConfigState{
 			Indent: " ",
 			// Extra deep spew.
 			DisableMethods: true,
 		}
 		t.Logf("import: %s", s.Sdump(d))
+		t.Fatalf("unexpected object: %#v", d.Image)
 	}
 }
 
@@ -588,11 +615,12 @@ func TestImageStreamImportRedHatRegistry(t *testing.T) {
 	rt, _ := kclient.TransportFor(&kclient.Config{})
 	importCtx := importer.NewContext(rt, nil).WithCredentials(importer.NoCredentials)
 
+	repositoryName := pulpRegistryName + "/rhel7"
 	// test without the client on the context
 	imports := &api.ImageStreamImport{
 		Spec: api.ImageStreamImportSpec{
 			Images: []api.ImageImportSpec{
-				{From: kapi.ObjectReference{Kind: "DockerImage", Name: "registry.access.redhat.com/rhel7"}},
+				{From: kapi.ObjectReference{Kind: "DockerImage", Name: repositoryName}},
 			},
 		},
 	}
@@ -617,15 +645,32 @@ func TestImageStreamImportRedHatRegistry(t *testing.T) {
 	imports = &api.ImageStreamImport{
 		Spec: api.ImageStreamImportSpec{
 			Images: []api.ImageImportSpec{
-				{From: kapi.ObjectReference{Kind: "DockerImage", Name: "registry.access.redhat.com/rhel7"}},
+				{From: kapi.ObjectReference{Kind: "DockerImage", Name: repositoryName}},
 			},
 		},
 	}
 	context := gocontext.WithValue(gocontext.Background(), importer.ContextKeyV1RegistryClient, dockerregistry.NewClient(20*time.Second, false))
 	importCtx = importer.NewContext(rt, nil).WithCredentials(importer.NoCredentials)
-	i = importer.NewImageStreamImporter(importCtx, 3, nil)
-	if err := i.Import(context, imports); err != nil {
-		t.Fatal(err)
+	err := retryWhenUnreachable(t, func() error {
+		i = importer.NewImageStreamImporter(importCtx, 3, nil)
+		if err := i.Import(context, imports); err != nil {
+			return err
+		}
+
+		errs := []error{}
+		for i, d := range imports.Status.Images {
+			fromName := imports.Spec.Images[i].From.Name
+			if d.Status.Status != unversioned.StatusSuccess {
+				errs = append(errs, fmt.Errorf("failed to import an image %s: %v", fromName, d.Status.Message))
+			}
+		}
+		return kerrors.NewAggregate(errs)
+	}, imageNotFoundErrorPatterns...)
+	if err != nil {
+		if strings.Contains(err.Error(), "x509: certificate has expired or is not yet valid") {
+			t.Skip("SKIPPING: due to expired certificate of %s: %v", pulpRegistryName, err)
+		}
+		t.Fatal(err.Error())
 	}
 
 	if imports.Status.Repository != nil {
@@ -635,9 +680,9 @@ func TestImageStreamImportRedHatRegistry(t *testing.T) {
 		t.Fatalf("unexpected response: %#v", imports.Status.Images)
 	}
 	d = imports.Status.Images[0]
-	if d.Image == nil || len(d.Image.DockerImageManifest) != 0 || d.Image.DockerImageReference != "registry.access.redhat.com/rhel7:latest" || len(d.Image.DockerImageMetadata.ID) == 0 || len(d.Image.DockerImageLayers) != 0 {
-		t.Errorf("unexpected object: %#v", d.Status)
+	if d.Image == nil || len(d.Image.DockerImageManifest) != 0 || d.Image.DockerImageReference != repositoryName+":latest" || len(d.Image.DockerImageMetadata.ID) == 0 || len(d.Image.DockerImageLayers) != 0 {
 		t.Logf("imports: %#v", imports.Status.Images[0].Image)
+		t.Fatalf("unexpected object: %#v", d.Status)
 	}
 }
 

